agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/FirstEntryPoint.java

```diff
@@ -162,7 +163,7 @@ private static void updateStatusFile(boolean success) {
       StatusFile.putValueAndWrite("AgentInitializedSuccessfully", success, startupLogger != null);
     } catch (Throwable t) {
       if (startupLogger != null) {
-        startupLogger.error("Error writing status.json", t);
+        startupLogger.error("Error writing status.json", t, MessageId.STATUS_FILE_RELATED_ERROR);
```

Comment:
I think this changes the meaning, when throwable is last it prints the stack trace

Also, I'm wondering now if it would be better to use MDC for storing/extracting the messageid

Reply:
MDC is also a good option